### ISSUE TITLE
get_rays get_ray_directions

### ISSUE THREAD

**Issue by wtj-zhong** [nerfstudio-project/nerfstudio#2539]
I'm new to NeRF and have recently become familiar with the original NeRF code. I'm currently working on NeRFStudio and am interested in finding out where in the NeRFStudio code the functions get_ray_directions(H, W, focus) and get_rays(directions, c2w) are implemented in the original NeRF code. We would greatly appreciate your help in finding these specific parts of the code. Thanks.
codes：
def get_ray_directions(H, W, focal):
    """
    Get ray directions for all pixels in camera coordinate.
    Reference: https://www.scratchapixel.com/lessons/3d-basic-rendering/
               ray-tracing-generating-camera-rays/standard-coordinate-systems

    Inputs:
        H, W, focal: image height, width and focal length

    Outputs:
        directions: (H, W, 3), the direction of the rays in camera coordinate
    """
    grid = create_meshgrid(H, W, normalized_coordinates=False)[0]
    i, j = grid.unbind(-1)
    # the direction here is without +0.5 pixel centering as calibration is not so accurate
    # see https://github.com/bmild/nerf/issues/24
    directions = \
        torch.stack([(i-W/2)/focal, -(j-H/2)/focal, -torch.ones_like(i)], -1) # (H, W, 3)

    return directions
